azure.go

```diff
@@ -120,87 +118,53 @@ func (p *azureProvider) Verifier() tokenVerifier {
 }
 
 func (p *azureProvider) ComputeClient(subscriptionID string) (computeClient, error) {
-	authorizer, err := p.getAuthorizer()
+	cred, err := az.NewClientSecretCredential(p.settings.TenantID, p.settings.ClientID, p.settings.ClientSecret, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	client := compute.NewVirtualMachinesClientWithBaseURI(p.settings.Environment.ResourceManagerEndpoint, subscriptionID)
-	client.Authorizer = authorizer
-	client.Sender = p.httpClient
-	client.AddToUserAgent(userAgent(p.settings.PluginEnv))
+	client, err := armcompute.NewVirtualMachinesClient(subscriptionID, cred, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// client := compute.NewVirtualMachinesClientWithBaseURI(p.settings.Environment.ResourceManagerEndpoint, subscriptionID)
```

Comment:
It doesn't look like the updated SDK allows the client to set `Sender` or to use `AddToUserAgent`. Not sure of the best approach as of yet, but still looking

Reply:
Thanks @fairclothjm for the helpful example! I believe we can successfully use the `userAgent` and `p.httpClient` fields as part of the new `transporter` interface which is passed in as `arm.ClientOptions`

I was also able to confirm that the update `armcompute` VM clients do not have a. `BaseURI` field, so it doesn't look like we can set a `BaseURI` anymore. The subscription ID forms a part of the URI for every service call in the updated SDK functions

Reply:
nice!